### PR TITLE
Fix varchar column display size

### LINKED ISSUE #10807: presto-jdbc: ResultSetMetaData.getColumnDisplaySize(int) returns 0 for varchar columns
1. Create table like "

create table test1 (

level_8 varchar(5000)

)

"

2. Then fetch the column size with :

resultSet = databaseMetadata.getColumns(null,null,"test1",null);

See results:

TABLE_CAT, TABLE_SCHEM, TABLE_NAME, COLUMN_NAME, DATA_TYPE, TYPE_NAME, COLUMN_SIZE, BUFFER_LENGTH, DECIMAL_DIGITS, NUM_PREC_RADIX, NULLABLE, REMARKS, COLUMN_DEF, SQL_DATA_TYPE, SQL_DATETIME_SUB, CHAR_OCTET_LENGTH, ORDINAL_POSITION, IS_NULLABLE, SCOPE_CATALOG, SCOPE_SCHEMA, SCOPE_TABLE, SOURCE_DATA_TYPE, IS_AUTOINCREMENT, IS_GENERATEDCOLUMN,

hive, labstuto, test1, level_8, -16, varchar(5000), 5000, 0, null, null, 2, null, null, null, null, 5000, 17, , null, null, null, null, null, null,

Notice the right column_size is defined as 5000.

3. Then Fetch the column size from the resultset metadata:

String q = null;
      q = "select * from " + "test1" + " limit 1";

      stmt = con.createStatement();
      rs = stmt.executeQuery(q);
      ResultSetMetaData rsMetadata = rs.getMetaData();
      for (int i = 1 ; i  < rsMetadata.getColumnCount()+1; i++){
        String name = rsMetadata.getColumnName(i);
        String type = Integer.toString(rsMetadata.getColumnType(i));
        String displaySize = Integer.toString(rsMetadata.getColumnDisplaySize(i));
        String precision = Integer.toString(rsMetadata.getPrecision(i));
        System.out.println("Name : ["+name+"], type: ["+type +"], display size: ["+displaySize+"], precision : ["+precision+"]");

      }

See the result:

Name : [level_8], type: [-16], display size: [0], precision : [0]

Expected Behavior:

rsMetadata.getColumnDisplaySize(i) should return 5000, instead of 0.

### PR DESCRIPTION
**Pull request by codope**
Fixes #10807

### CODE REVIEW

**Comment by codope** [comment]
@electrum  @dain @martint 
Could you please review this pull request? It will fix the discrepancy between column size returned by databaseMetadata.getColumns() and resultSetMetadata.getColumnDisplaySize().

File: presto-jdbc/src/main/java/com/facebook/presto/jdbc/ColumnInfo.java [hunk 1]
```diff
@@ -133,11 +134,23 @@ public static void setTypeInfo(Builder builder, TypeSignature type)
...
+                builder.setSigned(false);
+                builder.setPrecision(type.getParameters().get(0).getLongLiteral().intValue());
+                builder.setColumnDisplaySize(type.getParameters().get(0).getLongLiteral().intValue());
+                builder.setScale(0);
+                break;
             case "varchar":
-                builder.setSigned(true);
-                builder.setPrecision(VARCHAR_MAX);
+                builder.setSigned(false);
+                if (type.getParameters().get(0).getLongLiteral().intValue() > VarcharType.MAX_LENGTH) {
```
Comment (electrum) [inline presto-jdbc/src/main/java/com/facebook/presto/jdbc/ColumnInfo.java]: We can simplify this a bit using Math.max()

Comment (codope) [reply presto-jdbc/src/main/java/com/facebook/presto/jdbc/ColumnInfo.java]: Done.

**Comment by codope** [comment]
@electrum Is this good to be merged?

**Comment by ssainz** [comment]
@electrum  - Is ok to merge?

**Comment by ssainz** [comment]
@electrum - could you please merge the fix?

**Comment by tdcmeehan** [comment]
@codope 

* Can you please squash your commits?
* Can you please add test cases for this in `TestJdbcResultSet`

**Comment by electrum** [comment]
@codope @ssainz Apologies for the delay on this. I just merged the fix here: https://github.com/prestosql/presto/pull/615

**Comment by codope** [comment]
@electrum Thank you. Can you close the pr?

**Comment by tdcmeehan** [comment]
Merged, thanks!

**Pull request merged**